htdocs/core/lib/functions.lib.php

```diff
@@ -9999,6 +9999,8 @@ function printCommonFooter($zone = 'private')
 								// Add 'field required' class on closest td for all input elements : input, textarea and select
 								print 'jQuery(":input[name=\'' . $paramkey . '\']").closest("tr").find("td:first").addClass("fieldrequired");' . "\n";
 							}
+							// If we submit the cancel button we remove the required attributes
+							print 'jQuery("input[name=\'cancel\']").click(function() {jQuery("input, textarea, select").each(function(){this.removeAttribute(\'required\');});});'."\n";
```

Comment:
To help debug and understand what happen on any js code executed by a binded function, we must include a log into console. Can you add
console.log("We cick on cancel button so removed all required attribute")
just after 
`click(function() {`

Reply:
Log added as requested